samples/java_layout/src/main/java/io/customer/android/sample/java_layout/SampleApplication.java

```diff
@@ -28,22 +27,46 @@ public ApplicationGraph getApplicationGraph() {
     }
 
     private void initCustomerIOSDK() {
-        Map<String, String> configBundle;
+        Optional<CustomerIOSDKConfig> sdkConfigOptional;
         try {
-            configBundle = appGraph.getPreferencesDataStore().sdkConfig().blockingFirst();
+            Map<String, String> configBundle = appGraph.getPreferencesDataStore().sdkConfig().blockingFirst();
+            sdkConfigOptional = CustomerIOSDKConfig.fromMap(configBundle);
         } catch (Exception ignored) {
-            configBundle = new HashMap<>();
+            sdkConfigOptional = Optional.empty();
         }
-        final String apiKey = MapUtils.getOrElse(configBundle, CustomerIOSDK.API_KEY, BuildConfig.API_KEY);
-        final String siteId = MapUtils.getOrElse(configBundle, CustomerIOSDK.SITE_ID, BuildConfig.SITE_ID);
-        final String trackingApiUrl = MapUtils.getOrElse(configBundle, CustomerIOSDK.TRACKING_API_URL, null);
-        CustomerIO.Builder builder = new CustomerIO.Builder(siteId, apiKey, this);
-        builder.setLogLevel(CioLogLevel.DEBUG);
+        final CustomerIOSDKConfig sdkConfig;
+        if (sdkConfigOptional.isPresent()) {
+            sdkConfig = sdkConfigOptional.get();
+        } else {
+            sdkConfig = CustomerIOSDKConfig.getDefaultConfigurations();
+        }
+
+        CustomerIO.Builder builder = new CustomerIO.Builder(sdkConfig.getSiteId(), sdkConfig.getApiKey(), this);
+        if (sdkConfig.isDebugModeEnabled()) {
+            builder.setLogLevel(CioLogLevel.DEBUG);
+        }
+
+        final String trackingApiUrl = sdkConfig.getTrackingURL();
         if (!TextUtils.isEmpty(trackingApiUrl)) {
             builder.setTrackingApiURL(trackingApiUrl);
         }
+
+        final Integer bqSecondsDelay = sdkConfig.getBackgroundQueueSecondsDelay();
+        if (bqSecondsDelay != null) {
+            builder.setBackgroundQueueSecondsDelay(bqSecondsDelay);
+        }
+        final Integer bqMinTasks = sdkConfig.getBackgroundQueueMinNumOfTasks();
+        if (bqMinTasks != null) {
+            builder.setBackgroundQueueMinNumberOfTasks(bqMinTasks);
+        }
+
+        builder.autoTrackScreenViews(sdkConfig.isScreenTrackingEnabled());
+        builder.autoTrackDeviceAttributes(sdkConfig.isDeviceAttributesTrackingEnabled());
+
+        if (sdkConfig.isInAppEnabled()) {
+            builder.addCustomerIOModule(new ModuleMessagingInApp());
+        }
         builder.addCustomerIOModule(new ModuleMessagingPushFCM());
-        builder.addCustomerIOModule(new ModuleMessagingInApp());
         builder.build();
```

Comment:
Can you please elaborate what part of code makes you think it is difficult? It is mainly using options provided by our SDK. So IMO, this in fact is adding more help for customers on how to use these methods.

Reply:
I have an idea to suggest. Much of the code that got added in this PR is configuring the Customer.io SDK from the settings screen changes. 

If we were to abstract the code that configures the SDK from these settings screen options, this code would be closer to what my original comment is suggesting. 

[In the iOS sample apps, I did something similar](https://github.com/customerio/customerio-ios/blob/a0224143b12b9bcdb1548965f35fbd38c8b6d61a/Apps/CocoaPods-FCM/src/AppDelegate.swift#L24). Here is a code snippet of iOS: 

```swift
        CustomerIO.initialize(siteId: siteId, apiKey: apiKey, region: .US) { config in
            // Modify properties in the config object to configure the Customer.io SDK.

            config.logLevel = .debug // For all of our sample apps, we prefer to set debug logs to make our internal testing easier. You may not need to do this.

            // This line of code is internal to Customer.io for testing purposes. Do not add this code to your app.
            appSetSettings?.configureCioSdk(config: &config)
        }
```

The line of code that is relevant here is `appSetSettings?.configureCioSdk(config: &config)`. I tried to use only 1 line of code to perform the SDK configuration from settings screen changes. 

I like this solution because `appSetSettings?.configureCioSdk(config: &config)` is irrelevant to any customer that reads the sample app's source code. By us abstracting that code away from customers and leaving a comment telling customers they don't need to worry about that line of code, we are making the sample app's souce code more valuable for customers to read.

Reply:
```java
CustomerIO.Builder builder = new CustomerIO.Builder(sdkConfig.getSiteId(), sdkConfig.getApiKey(), this);

// Modify the Customer.io SDK through the Builder. 
builder.setLogLevel(CioLogLevel.DEBUG);

// Enable optional features of the SDK through enabling modules. 
builder.addCustomerIOModule(new ModuleMessagingPushFCM());

// This line of code is internal to Customer.io for testing purposes. Do not add this code to your app.
AppSettings.configureSdk(builder);

// Finally, build to finish initializing the SDK. 
builder.build();
```

Then, in new file `AppSettings`, 

```java
// AppSettings.java 

class AppSettings {

  static func configureSdk(builder) {
    // all of the code currently in Sampleapplication.java that modifies the builder
    Optional<CustomerIOSDKConfig> sdkConfigOptional;
        try {
            Map<String, String> configBundle = appGraph.getPreferencesDataStore().sdkConfig().blockingFirst();
            sdkConfigOptional = CustomerIOSDKConfig.fromMap(configBundle);
        } catch (Exception ignored) {
            sdkConfigOptional = Optional.empty();
        }
        final CustomerIOSDKConfig sdkConfig;
        if (sdkConfigOptional.isPresent()) {
            sdkConfig = sdkConfigOptional.get();
        } else {
            sdkConfig = CustomerIOSDKConfig.getDefaultConfigurations();
        }

final Integer bqSecondsDelay = sdkConfig.getBackgroundQueueSecondsDelay();
        if (bqSecondsDelay != null) {
            builder.setBackgroundQueueSecondsDelay(bqSecondsDelay);
        }
        final Integer bqMinTasks = sdkConfig.getBackgroundQueueMinNumOfTasks();
        if (bqMinTasks != null) {
            builder.setBackgroundQueueMinNumberOfTasks(bqMinTasks);
        }

        Boolean screenTrackingEnabled = sdkConfig.isScreenTrackingEnabled();
        if (screenTrackingEnabled != null) {
            builder.autoTrackScreenViews(screenTrackingEnabled);
        }
        Boolean deviceAttributesTrackingEnabled = sdkConfig.isDeviceAttributesTrackingEnabled();
        if (deviceAttributesTrackingEnabled != null) {
            builder.autoTrackDeviceAttributes(deviceAttributesTrackingEnabled);
        }

        if (Boolean.TRUE.equals(sdkConfig.isInAppEnabled())) {
            builder.addCustomerIOModule(new ModuleMessagingInApp());
        }

return builder; 
  }

}
```

Reply:
Thanks for the code sample, I've updated it after some tweaks. I might be making more changes to this in future PRs so any further change can be discussed there.